### PR TITLE
fixed Engine::getVersion not passing agent to new Engine constructor

### PR DESCRIPTION
**Pull request by shkkmo**
Also added an isset() check to prevent PHP notices due to undefined array offset

See bug report:
https://pear.php.net/bugs/bug.php?id=21034

### CODE REVIEW

**Comment by gauthierm** [comment]
Please add a test for this fix.

**Comment by shkkmo** [comment]
I do not know how to create a test that mocks having the agent executable in a non-standard location.

My pull request fixes an oversight made in this commit when you added the 'agent' option to the engine constructor:
https://github.com/pear/Crypt_GPG/commit/955c950353ddf067bbb1e173a673f6293f5a6e1a

File: Crypt/GPG/Engine.php [hunk 1]
```diff
@@ -1662,7 +1663,7 @@ private function _openSubprocess()
             );
 
             $agentInfo             = explode(' ', $agentInfo, 3);
-            $this->_agentInfo      = $agentInfo[2];
+            $this->_agentInfo      = isset($agentInfo[2])?$agentInfo[2]:null;
```
Comment (alecpl) [inline Crypt/GPG/Engine.php]: Please, fix code style here. Add some spaces.

**Comment by alecpl** [comment]
Let's go without a test, it's an obvious fix and hard to test.

**Pull request merged**